### DISCUSSION TITLE
Tabout doesn't work. Is it intended?

### DISCUSSION THREAD

**Discussion by txtyash** [helix-editor/helix#1209] [General]
Helix does auto-pairs by default but to move out of the pair I have to press the closing `'")]}` part of the pair which makes auto-pairs pointless and rather annoying. To tackle this issue editors like vs-code have tabout feature where you hit the `tab` key to move out of a pair whereas neovim has a plugin called tabout.nvim for this.

I tested tabout but doesn't seem to work in helix.

**Comment by CBenoit** [comment]
I didn't know about this. Looks interesting!

**Comment by dead10ck** [comment]
This is the first I've heard of this, and it is a very interesting idea. Judging by the demo animations in [tabout.nvim](https://github.com/abecodes/tabout.nvim), it looks like this could be even further generalized to a command that moves the cursor to the next end of a syntax node range. It could literally just walk the syntax tree. If the cursor is in the middle of a node, move it to the end; if at the end of a syntax node, move it to the next node's beginning, etc. And also one that moves backward.

**Comment by kirawi** [comment]
Is it different from https://github.com/helix-editor/helix/issues/956?

**Reply by dead10ck** [reply]
I think so. From what I'm gathering, the proposed feature is to move only forward or backward, whereas mm bounces bounces back and forth inside the same pair.

**Reply by txtyash** [reply]
Tabout actually takes you out of `'' or "" or () ...` in insert mode. For
example if you want to write `print("something");` then in simpler steps:
- First you write `print`
- Then you write `(` which get auto-paired to `()`
- Now your current state is `print(|)`(`|` represents cursor position)
- You write `"`: `print("|")`(`|` represents cursor position)
- Now you type `something`: `print("something|")`(`|` represents cursor position)
- Notice that you are in insert mode, at the end of `something` and you have to
move beyond `")`. Here you can either get into normal mode, go to the end, get
in insert mode and type `;` to complete the task or you can type `")` which will take you to the end so you can type `;` to complete the task.
- Instead of the two ways given above you can just press `tab` twice at
`print("something|")`(`|` represents cursor position) to move to the end
`print("something")|`, still in insert mode to type `;` and complete the task.

**Reply by txtyash** [reply]
> Is it different from #956?

yes
